### PR TITLE
Add fast lookup feature spec

### PR DESCRIPTION
**Pull request by sunxiaoguang**
Closes: #129

### CODE REVIEW

File: docs/specs/lookup.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Product Requirements Specification
+
+Implement fast lookup via primary key or secondary indices
+
+## Problem:
+
+There are use cases within big-data scenarios where fast lookup with index are beneficial. 
+Especially in streaming processing, lookup latency have a significant impact to overall
+throughput. TiBigData implements full table scan only at this time, which makes it suboptimal to
```
Comment (xuanyu66) [inline docs/specs/lookup.md:9]: At this time, TiBigData only implements full table scan

File: docs/specs/lookup.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+workloads that requires low latency. Therefore, flink-tidb-connector utilize JDBC lookup table 
+source instead. Since Flink, like other big-data computation frameworks, is at computing layer by
+itself. Sending lookup request to TiDB server which is at computing layer as well is redundant and 
+waste of resources. Lookup with TiDB server contributes to growing of latency with extra hops in
```
Comment (xuanyu66) [inline docs/specs/lookup.md:13]: a waste of resources

File: docs/specs/lookup.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+access path.
+
+## Goals:
+
+Implement primary key and secondary index based lookup in TiBigData. Decreases lookup latency with
```
[the remark this answers is not in the document]
Comment (xuanyu66) [reply docs/specs/lookup.md:18]: It decreases

File: docs/specs/lookup.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+the best efforts and saves unnecessary resources waste for load balancers and TiDB servers.
```
Comment (xuanyu66) [inline docs/specs/lookup.md:19]: the unnecessary waste of resources

**Review by xuanyu66 — CHANGES_REQUESTED** [review]

File: docs/specs/lookup.md (hunk 4 above)
Comment (sunxiaoguang) [reply docs/specs/lookup.md]: Changed from `waste of resources` to `costs`.

**Comment by xuanyu66** [comment]
LGTM

**Review by xuanyu66 — APPROVED** [review]

**Review by shiyuhang0 — APPROVED** [review]

**Pull request closed without merging**